test/Microsoft.AspNet.Diagnostics.Tests/ElmPageTest.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNet.FileProviders;
+using Xunit;
+
+namespace Microsoft.AspNet.Diagnostics
+{
+    public class ElmPageTest
+    {
+        [Theory]
+        [InlineData("TestFiles/SourceFile.txt")]
+        [InlineData(@"TestFiles\SourceFile.txt")]
+        public void DisplaysSourceCodeLines_ForAbsolutePaths(string path)
+        {
+            // Arrange
+            var rootPath = Directory.GetCurrentDirectory();
+            var absoluteFilePath = Path.Combine(rootPath, path);
+            // PhysicalFileProvider handles only relative paths but we fall back to work with absolute paths too
+            var provider = new PhysicalFileProvider(rootPath);
+
+            // Act
+            var stackFrame = ErrorPageMiddleware.GetStackFrame(
+                "func1",
+                absoluteFilePath,
+                lineNumber: 10,
+                fileProvider: provider,
+                sourceCodeLineCount: 6);
+
+            // Assert
+            // Lines 4-17 (inclusive) is the code block
+            Assert.Equal(4, stackFrame.PreContextLine);
+            Assert.Equal<string>(GetCodeLines(4, 6), stackFrame.PreContextCode);
+            Assert.Equal("Line10", stackFrame.ContextCode.First());
+            Assert.Equal<string>(GetCodeLines(11, 6), stackFrame.PostContextCode);
+        }
+
+        [Theory]
+        [InlineData("TestFiles/SourceFile.txt")]
+        [InlineData(@"TestFiles\SourceFile.txt")]
+        public void DisplaysSourceCodeLines_ForRelativePaths(string relativePath)
+        {
+            // Arrange
+            var rootPath = Directory.GetCurrentDirectory();
+            var provider = new PhysicalFileProvider(rootPath);
+
+            // Act
+            var stackFrame = ErrorPageMiddleware.GetStackFrame(
+                "func1",
+                relativePath,
+                lineNumber: 10,
+                fileProvider: provider,
+                sourceCodeLineCount: 6);
+
+            // Assert
+            // Lines 4-17 (inclusive) is the code block
+            Assert.Equal(4, stackFrame.PreContextLine);
+            Assert.Equal<string>(GetCodeLines(4, 6), stackFrame.PreContextCode);
+            Assert.Equal("Line10", stackFrame.ContextCode.First());
+            Assert.Equal<string>(GetCodeLines(11, 6), stackFrame.PostContextCode);
+        }
+
+        [Theory]
+        [InlineData("TestFiles/EmbeddedSourceFile.txt")]
+        //[InlineData(@"TestFiles\EmbeddedSourceFile.txt")]
```

Comment:
@victorhurdugaci This kind of relative path works when using PhysicalFileProvider but does not when using Embedded...since in any case we would be replacing `/` with a `.`, should we even handle `\` too?